### PR TITLE
Reworked updater so that it can pull multiple branches, some fixes

### LINKED ISSUE #88: Updater needs to support validation of non-master branches of target repositories

### PR DESCRIPTION
**Pull request by renatav**
## Description (e.g. "Related to ...", etc.)

- Support validation of non-master branches of target repositories. Updater fetches and validates all branches of a repository specified in the corresponding target file of the authentication repository
- Added tests for updating repositories with multiple branches
- Discovered and fixed some bugs while creating the new test repositories:
  - fixed `update_target_repos_from_repositories_json` 
  - added an option to store information about the current branch when updating targets files
  - added missing `scheme` parameter in a few places

Closes #88 

## Code review checklist (for code reviewer to complete)

- [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR)
- [ ] Title summarizes what is changing
- [ ] Commit messages are meaningful (see [this][commit messages] for details)
- [ ] Tests have been included and/or updated, as appropriate
- [ ] Docstrings have been included and/or updated, as appropriate
- [ ] Changelog has been updated, as needed (see [CHANGELOG.md][changelog])

[changelog]: https://github.com/openlawlibrary/taf/blob/master/CHANGELOG.md
[commit messages]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: taf/cli.py [hunk 1]
```diff
@@ -51,19 +51,20 @@ def add_signing_key(repo_path, role, pub_key_path):
 @click.option('--targets-dir', default='targets', help='Directory where the target '
               'repositories are located')
 @click.option('--namespace', default=None, help='Namespace of the target repositories')
-def update_repos_from_fs(repo_path, targets_dir, namespace):
+@click.option('--add-branch', default=True, is_flag=True, help="Whether to add name of current branch to target file")
```
Comment (danixeee) [inline taf/cli.py]: `add_branch` will be `True` even if `--add-branch` is not passed.

File: taf/cli.py [hunk 2]
```diff
@@ -51,19 +51,20 @@ def add_signing_key(repo_path, role, pub_key_path):
...
+    developer_tool.update_target_repos_from_fs(repo_path, targets_dir, namespace, add_branch)
 
 
 @cli.command()
 @click.option("--repo-path", default="repository", help="Location of the repository")
 @click.option('--targets-dir', default='targets', help='Directory where the target '
               'repositories are located')
-@click.option('--namespace', default=None, help='Namespace of the target repositories')
-def update_repos_from_repositories_json(repo_path, targets_dir, namespace):
+@click.option('--add-branch', default=True, is_flag=True, help="Whether to add name of current branch to target file")
```
Comment (danixeee) [inline taf/cli.py]: https://github.com/openlawlibrary/taf/pull/91/files#r363304213

File: taf/developer_tool.py [hunk 3]
```diff
@@ -185,27 +186,38 @@ def update_target_repos_from_fs(repo_path, targets_directory, namespace=None):
...
-        _update_target_repos(repo_path, targets_dir, target_repo_path)
+    auth_repo_path = Path(repo_path).resolve()
+    auth_repo_targets_dir = auth_repo_path / TARGETS_DIRECTORY_NAME
+    repositories_json = json.loads(
+        Path(auth_repo_targets_dir / "repositories.json").read_text()
+    )
+    targets_directory = Path(targets_directory).resolve()
+    for repo_name in repositories_json.get("repositories"):
+        target_repo_path = targets_directory / repo_name
+        namespeace_and_name = repo_name.rsplit("/", 1)
```
Comment (danixeee) [inline taf/developer_tool.py]: typo

File: taf/developer_tool.py [hunk 4]
```diff
@@ -503,7 +514,7 @@ def generate_repositories_json(
         url = target_repo.get_remote_url()
         if url is None:
             if targets_relative_dir is not None:
-                url = Path(target_repo.repo_path).relative_to(targets_relative_dir)
+                url = Path(os.path.relpath(target_repo.repo_path, targets_relative_dir))
```
Comment (danixeee) [inline taf/developer_tool.py]: Was this a windows issue with paths?

File: taf/updater/updater.py [hunk 5]
```diff
@@ -169,15 +170,22 @@ def update_named_repository(
         repositories = repositoriesdb.get_deduplicated_repositories(
             users_auth_repo, commits
         )
-        repositories_commits = users_auth_repo.sorted_commits_per_repositories(commits)
+        # TODO
```
Comment (danixeee) [inline taf/updater/updater.py]: Is this obsolete?

File: taf/auth_repo.py [hunk 6]
```diff
@@ -114,10 +128,13 @@ def target_commits_at_revisions(self, commits):
                     # we only care about repositories
                     continue
                 try:
-                    target_commit = self.get_json(
+                    # TODO get branch here as well
```
Comment (danixeee) [inline taf/auth_repo.py]: Is this obsolete?

File: taf/developer_tool.py (hunk 4 above)
Comment (renatav) [reply taf/developer_tool.py]: Not sure if it is Windows specific problem, but `relative_to` does not work as well as `relpath`. I remember having that issue before

File: taf/updater/updater.py (hunk 5 above)
Comment (renatav) [reply taf/updater/updater.py]: Yes

File: taf/auth_repo.py (hunk 6 above)
Comment (renatav) [reply taf/auth_repo.py]: Yes

**Comment by renatav** [comment]
@danixeee Let's keep that flag for now and worry about it when we do the big cli rework

**Pull request merged**